daemon-tests/src/flow.rs

```diff
@@ -39,15 +40,9 @@ pub async fn next_maker_offers(
     Ok((a?, b?))
 }
 
-pub async fn is_next_offers_none(
-    rx: &mut OffersFeed,
-    contract_symbol: &ContractSymbol,
-) -> Result<bool> {
-    let mut rx = match contract_symbol {
-        ContractSymbol::BtcUsd => rx.btc_usd.clone(),
-        ContractSymbol::EthUsd => rx.eth_usd.clone(),
-    };
-    let maker_offers = next(&mut rx).await?;
+// TODO: what is this function even doing? only called from the tests and the response is ignored.
```

Comment:
This function ensures that (initially) the value consumed from the feed is `None`.
This is used for sanity in the tests. I think we should keep it, but am open to discussing better solutions.

Reply:
lets keep it then.

Reply:
I agree that it wasn't working as intended but I think what we should do is simply bubble up an error if the condition is not met.

That means the return type would be `Result<()>` and the caller can just unwrap since we're in a test, and ignore the `Ok(())`.

I would vote against `assert`ing because that should really be used to verify whatever the test is actually testing, rather than to check if there's  a bug in the implementation of the test.